src/js/components/DataTable/DataTable.js

```diff
@@ -173,6 +176,40 @@ const DataTable = ({
     }
   }, [footerRef, headerRef, placeholder]);
 
+  useEffect(() => {
+    const getPinnedOffset = () => {
+      const pinnedProperties = columns
+        .map(pinnedColumn => pinnedColumn.pin && pinnedColumn.property)
+        .filter(n => n);
+
+      const pinnedColumnsProperties = {};
+      const firstRowCells = headerRef?.current?.firstChild?.cells || [];
```

Comment:
Just for my own understanding, is the `?` syntax the equivalent of saying the following?
```
headerRef && headerRef.current && headerRef.current.firstChild ...
```

Reply:
Pretty much that. That was introduced as code was breaking when trying to access data from `headerRef.current` while it was undefined. Just realized this is kinda gone once we moved the logic to an useEffect hook. I'd suggest we go with one of those two approaches, in that case:

`headerRef.current.firstChild.cells || []`
or
`headerRef.current.firstChild.cells ?? []`